bindings/py/tests/algorithms/SDR_test.py

```diff
@@ -0,0 +1,242 @@
+        A = SDR((103,))
+        B = SDR((100, 100, 1))
+
+        A.dense
+        # Test is the same buffer every time
+        A.dense[0] = 1
+        A.dense[99] = 1
+        assert(A.dense[0] + A.dense[99] == 2)
+        # Test modify in-place
+        A.setDenseInplace()
+        assert(set(A.flatSparse) == set((0, 99)))
+        # Test dense dimensions
+        assert(B.dense.shape == (100, 100, 1))
+        # No crash with dimensions
+        B.dense[0, 0, 0] += 1
+        B.dense[66, 2, 0] += 1
+        B.dense[99, 99, 0] += 1
+        B.dense = B.dense.reshape(-1)
+        # Test wrong dimensions assigned
+        C = SDR((2, 4, 5, 1,1,1,1, 3))
+        C.randomize( .5 )
+        try:
+            A.dense = C.dense.reshape(-1)
+        except RuntimeError:
+            pass
+        else:
+            self.fail()
+
+    def testFlatSparse(self):
+        A = SDR((103,))
+        B = SDR((100, 100, 1))
+
+        A.flatSparse
+        B.flatSparse = [1,2,3,4]
+        assert(all(B.flatSparse == np.array([1,2,3,4])))
+
+        # Test wrong dimensions assigned
+        C = SDR((2, 4, 5, 1,1,1,1, 3))
+        C.randomize( .5 )
+        try:
+            A.flatSparse = C.flatSparse
+        except RuntimeError:
+            pass
+        else:
+            self.fail()
+
+    def testSparse(self):
+        A = SDR((103,))
+        B = SDR((100, 100, 1))
+        C = SDR((2, 4, 5, 1,1,1,1, 3))
+
+        A.sparse
+        B.sparse = [[0, 55, 99], [0, 11, 99], [0, 0, 0]]
+        assert(B.dense[0, 0, 0]   == 1)
+        assert(B.dense[55, 11, 0] == 1)
+        assert(B.dense[99, 99, 0] == 1)
+        C.randomize( .5 )
+        assert( len(C.sparse) == len(C.dimensions) )
+
+        # Test wrong dimensions assigned
+        C = SDR((2, 4, 5, 1,1,1,1, 3))
+        C.randomize( .5 )
+        try:
+            A.sparse = C.sparse
+        except RuntimeError:
+            pass
+        else:
+            self.fail()
+
+    def testSetSDR(self):
+        A = SDR((103,))
+        B = SDR((103,))
+        A.flatSparse = [66]
+        B.setSDR( A )
+        assert( B.dense[66] == 1 )
+        assert( B.getSum() == 1 )
+        B.dense[77] = 1
+        B.setDenseInplace()
+        A.setSDR( B )
+        assert( set(A.flatSparse) == set((66, 77)) )
+
+        # Test wrong dimensions assigned
+        C = SDR((2, 4, 5, 1,1,1,1, 3))
+        C.randomize( .5 )
+        try:
+            A.setSDR(C)
+        except RuntimeError:
+            pass
+        else:
+            self.fail()
+
+    def testGetSum(self):
+        A = SDR((103,))
+        assert(A.getSum() == 0)
+        A.dense.fill(1)
+        A.setDenseInplace()
+        assert(A.getSum() == 103)
+
+    def testGetSparsity(self):
+        A = SDR((103,))
+        assert(A.getSparsity() == 0)
+        A.dense.fill(1)
+        A.setDenseInplace()
+        assert(A.getSparsity() == 1)
+
+    def testOverlap(self):
+        A = SDR((103,))
+        B = SDR((103,))
+        assert(A.overlap(B) == 0)
```

Comment:
does `overlap` alter A, or B? If not, I'd prefer name `getOverlap` for both py, c++

Reply:
`A.overlap(B)` does not change the value of either `A` or `B` however it does require that both be non-constant.

Reply:
I'd say get* would be more natural in that case

Reply:
👍 thanks